programs/manifest/src/state/market.rs

```diff
@@ -1167,6 +1164,115 @@ impl<
                 #[cfg(feature = "certora")]
                 add_to_orderbook_balance(fixed, dynamic, current_maker_order_index);
                 remaining_base_atoms = BaseAtoms::ZERO;
+            }
+
+            // Place the reverse order if the maker was a reverse order type.
+            // This is non-trivial because in order to prevent tons of orders
+            // filling the books on partial fills, we coalesce on top of book.
```

Comment:
Do we only coalesce orders on ToB?

Reply:
Yes. We could do a search beyond if we think that is necessary. This was intended as a hack since it assumes that the spread will be small and move to top of book. if that is not the case in practice, we will either get lots of orders, or need to do a search for coalescing